frontend/catalyst/cuda/primitives/__init__.py

```diff
@@ -396,11 +396,7 @@ def cudaq_sample_impl(kernel, *args, shots_count=1000):
     a_dict = cudaq.sample(kernel, *args, shots_count=shots_count)
     aggregate = []
     for bitstring, count in a_dict.items():
-        # It is technically a bit array
-        # So we should use int(bit)
-        # But in Catalyst, these are floats.
-        # So we use floats.
-        bitarray = [float(bit) for bit in bitstring]
+        bitarray = [int(bit) for bit in bitstring]
```

Comment:
@erick-xanadu Could you double check the changes to the CUDA integration, I wasn't sure about those.